### DISCUSSION TITLE
How can I render MudButton in EditObject Dialog Meta Configuration

### DISCUSSION THREAD

**Discussion by Sharafudeen** [fgilde/MudBlazor.Extensions#87] [General]
How can I render MudButton in EditObject Dialog Meta Configuration

**Comment by fgilde** [comment]
Depends on what you plan do do with this button and if it's related to a property in the model. General with 
'''
meta.Property(a=> a.Property).RenderWith<MudButton>()
'''
But then you need to pass the click event handler in parameters with EventCallbackFactory.create 

If I find time I will create a sample for you tomorrow currently I only have a mobile phone here

**Comment by Sharafudeen** [comment]
@fgilde Can you give example for rendering the MudButton with OnClick Event method?

**Comment by fgilde** [comment]
Sure.  In general this is what you need to do

```c#
    private void Configure(ObjectEditMeta<SampleModel> meta)
    {
        meta.Property(x => x.WhatEverProp).RenderWith<MudButton>(button =>
        {
            button.ChildContent = DynamicChildContent("Click me");
            button.OnClick = EventCallback.Factory.Create<MouseEventArgs>(this, () =>
            {
                JsRuntime.AlertAsync("Button Clicked");
            });
        });
    }

    private static RenderFragment DynamicChildContent(string s)
    {
        return builder =>
        {
            var idx = 0;
            builder.OpenElement(idx++, "p");
            builder.AddContent(idx, s);
            builder.CloseElement();
        };
    }
```
I created a sample for you here, also with the sample how to acess files for upload etc.

https://try.mudex.org/snippet/mEGyEAmMHGqGKEow

**Reply by Sharafudeen** [reply]
@fgilde This is fine.. button OnClick can set outside click event method?

**Reply by fgilde** [reply]
sure

**Reply by Sharafudeen** [reply]
Example please.. :)

**Reply by fgilde** [reply]
https://try.mudex.org/snippet/GEQouhumydarUQRR
